packages/react-components/react-persona/Spec.md

```diff
@@ -1,63 +1,212 @@
+  root: Slot<'div'>;
+
+  /**
+   * Avatar to display.
+   */
+  avatar?: Slot<typeof Avatar>;
+
+  /**
+   * PresenceBadge to display.
+   */
+  presence?: Slot<typeof PresenceBadge>;
+
+  /**
+   * Primary text to display.
+   */
+  primaryText?: Slot<'span'>;
+
+  /**
+   * Secondary text to display.
+   */
+  secondaryText?: Slot<'span'>;
+
+  /**
+   * Tertiary text to display.
+   */
+  tertiaryText?: Slot<'span'>;
+
+  /**
+   * Quaternary text to display.
+   */
+  quaternaryText?: Slot<'span'>;
+};
+
+/**
+ * Persona Props
+ */
+export type PersonaProps = ComponentProps<PersonaSlots> & {
+  /**
+   * Sizing type to use.
+   *
+   * `fixed`: Text lines adjust to content's size.
+   * `scaled`: Content adjusts its size based on the number of text lines used.
+   */
+  sizing?: 'fixed' | 'scaled';
+
+  /**
+   * Position the Avatar/PresenceBadge will be rendered in.
+   */
+  position?: 'start' | 'center' | 'end';
+};
+```
 
 ## Structure
 
-- _**Public**_
+To avoid the [issue](https://github.com/microsoft/fluentui/issues/23386) v8 has, a css grid will be used instead of a flexbox that requires a general wrapper and a text container wrapper.
+
+- _**CSS Grid**_
+
+  - ![Start](./etc/images/grid_start.png)
+  - ![Center](./etc/images/grid_center.png)
+  - ![End](./etc/images/grid_end.png)
+
 - _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
 
-## Migration
+```jsx
+return (
+  <slots.root {...slotProps.root}>
+    {slots.avatar && <slots.avatar {...slotProps.avatar} />}
+    {slots.presence && <slots.presence {...slotProps.presence} />}
+    {slots.primaryText && <slots.primaryText {...slotProps.primaryText} />}
+    {slots.secondaryText && <slots.secondaryText {...slotProps.secondaryText} />}
+    {slots.tertiaryText && <slots.tertiaryText {...slotProps.tertiaryText} />}
+    {slots.quaternaryText && <slots.quaternaryText {...slotProps.quaternaryText} />}
+  </slots.root>
+);
+```
+
+- _**DOM**_
+
+```html
+<div class="fui-Persona">
+  <div {/* Avatar, PresenceBadge, or Avatar with PresenceBadge */} />
+  <span class="fui-Persona__primaryText">Primary Text</span>
+  <span class="fui-Persona__secondaryText">Secondary Text</span>
+  <span class="fui-Persona__tertiaryText">Tertiary Text</span>
+  <span class="fui-Persona__quaternaryText">Quaternary Text</span>
+</div>
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+## Migration
 
-- _Migration from v8_
-- _Migration from v0_
+See [MIGRATION.md](./MIGRATION.md) for details.
 
 ## Behaviors
 
 _Explain how the component will behave in use, including:_
 
 - _Component States_
+  - There are no states for this component, it's a visual representation.
 - _Interaction_
   - _Keyboard_
+    - Doesn't receive focus.
   - _Cursor_
+    - Doesn't interact with cursor.
   - _Touch_
+    - Doesn't interact with touch.
   - _Screen readers_
+    - It first focuses on the media and then goes through each text line available.
```

Comment:
To clarify, this is just describing the behavior of a screen reader right? Persona isn't doing anything here.

Reply:
correct!

Reply:
Hmm thinking more about this: in the `textPosition="before"` case, should the reading order follow the visual order: text then avatar? That would best be done by changing the DOM order of the slots, which is also part of my recommendation in this comment on the implementation PR: https://github.com/microsoft/fluentui/pull/24749#discussion_r971454281